### DISCUSSION TITLE
[Feature]  Search photos with only the selected persons

### DISCUSSION THREAD

**Discussion by dvelayos** [immich-app/immich#10373] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Now, you can search by person, and the search results will show all the photos where this person appears (possibly with others). However, it would be great if you could search for photos featuring ONLY this person. This functionality could also apply to more than one person; for example, I want to search for all the photos of my daughters, and only my daughters. This could be added to the interface with a checkbox labeled "Only these persons". 

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by blublub** [comment]
thats already implemented
![grafik](https://github.com/immich-app/immich/assets/10511888/63050e72-afbb-417d-aefb-82c9ce5ab636)

**Comment by dvelayos** [comment]
Maybe my explanation is not clear enough; i know that you can filter by person, i want to filter photos by person AND without all the other recognized persons, i mean, the photos with this person completely alone (or, if you select two persons, the photos with those two persons and not other persons)

if i search selecting a person, i see all the photos on which this person appears, but there are other recognized persons on the photos. If you search "alone" or "portrait" by context, it do the trick, more or less

also, maybe it would be useful to split this functionality in two checks; "not other named persons" and "no other persons (named or not)"

**Comment by blublub** [comment]
ah ok - now I get it.

So a feature to avoid "anyone else"

**Comment by imohitkr** [comment]
When I went on a trip with my friends, I took a lot of photos. While I want to keep the group photos, I don't want to keep the individual photos or group photos where I do not happen to be present. It would be helpful to have a feature that allows me to search for and view only photos that exactly match the people's selection, not less or more. This way, the search results wouldn't display photos with more or fewer people than I specified. And we can bulk delete them.

**Comment by apollo79** [comment]
I am also looking for this functionality. It would be very helpful to find e. g. specific images for a damily album.

**Comment by grillop** [comment]
I have been wanting this capability for a long time.  I put up immich power tools as it has many features, one of which supposedly would handle this.  Unfortunately it is part of the larger natural language AI query that has a bunch of issues.
sigh ... for those of you who are a bit techy, and don't mind messing  with a little with python... i decided to write my own interface to handle this really just for you home system.  It is a lightweight python script powered using streamlit.  truthfully, took very little time with all of the AI help available - so I'm not taking full credit here, but i've been tweaking it.

it sits on the same server as immich and has access to the database.  So, it's pretty simple.  You get a dropdown of all your named people.  You can choose any number of them, and issue a search.  It will bring back matching thumbs.  You have a couple of options. A checkbox that says, only them.  Also, you can choose sort old or new.  If you click on a thumb it will invoke immich to display the full image.

It's pretty fast.. it doesn't jam all of the thumbs to the browser, it pages it with an internal paging cache - so it is extremely fast - given that it is straight sql, and stores pointers.  It paginates, next/previous  or at the bottom, pages you can jump to.

anyway, i'm not into posting code or anything - if you want to mess with it, DM me and i'll put it together in a reasonable package.  Consists of a script, compose yaml, and a Docker file that when invoked will recreate and deploy the container.  You'll be on your own ... i mean, i can clarify what some of the variables are ... but, i'm not planning to post it on github or anything, at least not now.


<img width="1194" height="791" alt="OnlyThose" src="https://github.com/user-attachments/assets/104cae37-21cf-4faf-ba41-b2834c8f0487" />

**Comment by seanbrown-com** [comment]
I’d also love to have this feature implemented. For anyone that takes photos with their family, there are many photos of the whole family, but if I only want photos of, say, my wife and me, or my two kids, or me and one kid, me and both kids, my wife and one kid, wife and both kids, there’s no way to achieve that. Also, we have many photos of the larger family or with groups of friends that also come up in a search for just two or three people. This would be an invaluable feature!

**Reply by seanbrown-com** [reply]
I’m a software engineer, but I’ve never worked on this project. How likely would a feature like this be approved if I submitted a PR?

**Reply by seanbrown-com** [reply]
Example feature request:

Feature Request: Exact People-Only Photo Filtering

User Need:
Users need a way to find photos that contain exactly a selected set of people and exclude any photos with additional detected people. For example, selecting “Sid” and “Nancy” should return only photos where both are present and no other recognized or detected people appear.

Proposed Solution:
Add a filter option such as “Only these people” or “Exclude other detected people” to person/face search. This filter would limit results to photos containing the selected individuals and no others, with an optional setting to include or exclude unknown/unassigned detected faces. This would make it easier to create albums and locate specific family photos.

**Comment by vg77-it** [comment]
Are there any updates regarding this feature?

I would also really love to have this functionality. A kind of AND/OR filter for faces, so that when selecting one or more people, you have the option to view them either alone or together with others.

**Reply by bo0tzz** [reply]
https://justinmayer.com/posts/any-updates/

**Comment by yanshay** [comment]
I think I found a partial solution for this. Select the people you want in the picture and then in context write how many people. 
So for example John, Alice and in context "two people".
For some reason it is not perfect, and I sometime get photos with 3 or 4 people. But I don't get those large group photos.
Would be interesting to understand why counting is not accurate, maybe its because it is AI embeddings and not real face count?
